libs/libGamedata/gameData/CampaignDescription.cpp

```diff
@@ -37,12 +37,28 @@ CampaignDescription::CampaignDescription(const kaguya::LuaRef& table)
     luaData.checkUnused();
 }
 
-boost::filesystem::path CampaignDescription::getLuaFilePath(const std::string& mapName) const
+size_t CampaignDescription::getNumMaps()
 {
-    return (RTTRCONFIG.ExpandPath(luaFolder) / mapName).replace_extension("lua");
+    return mapNames.size();
 }
 
-boost::filesystem::path CampaignDescription::getMapFilePath(const std::string& mapName) const
+std::string CampaignDescription::getMapName(const size_t idx) const
 {
-    return RTTRCONFIG.ExpandPath(mapFolder) / mapName;
+    if(idx < mapNames.size())
+        return mapNames[idx];
+    return std::string();
+}
+
+boost::filesystem::path CampaignDescription::getLuaFilePath(const size_t idx) const
+{
+    if(idx < mapNames.size())
+        return (RTTRCONFIG.ExpandPath(luaFolder) / mapNames[idx]).replace_extension("lua");
+    return boost::filesystem::path();
```

Comment:
Better throw an exception here and in the others

Reply:
Make no check anymore at use at(idx) on vector container so a exception is thrown automatically when out of range.